### PR TITLE
In step-77, also talk about the other nonlinear solver packages we support.

### PR DESCRIPTION
**Pull request by bangerth**
In relation to #15065 and #14332.

### CODE REVIEW

**Review by marcfehling — APPROVED** [review]

File: examples/step-77/doc/intro.dox [hunk 1]
```diff
@@ -155,6 +155,18 @@ provides a vast amount of finite-element functionality, %SUNDIALS'
...
+@note While this program uses SUNDIAL's KINSOL package as the engine to
+  solve nonlinear problems, KINSOL is not the only option you have.
+  deal.II also has interfaces to PETSc's SNES collection of algorithms
+  (see the PETScWrappers::NonlinearSolver class) as well as to
+  the Trilinos NOX package (see the TrilinosWrappers::NOXSolver class)
+  that provide not only very similar functionality, but also a largely
+  identical interface. If you have installed a version of deal.II that
+  is configured to use either PETSc or Trilinos, but not SUNDIALS,
+  then it is not too difficult to switch this program to use either
+  of the former two packages instead: Basically everything that we
```
Comment (stefanozampini) [inline examples/step-77/doc/intro.dox:167]:
```suggestion
  of the former two packages instead: basically everything that we
```

Comment (bangerth) [reply examples/step-77/doc/intro.dox:167]: I don't think this is actually incorrect: The sentence that follows the colon is a complete sentence, and can consequently be capitalized. (See what I'm doing here? ;-)

I agree that the first word should not be capitalized if what follows is not a complete sentence: a list, or half sentence, for example.

Comment (stefanozampini) [reply examples/step-77/doc/intro.dox:167]: american vs british english (and the rest of the other languages based on the alphabet ...)

**Comment by bangerth** [comment]
Ping?

**Review by stefanozampini — APPROVED** [review]

**Review by peterrum — APPROVED** [review]

**Pull request merged**